pkg/services/dashboards/database/database.go

```diff
@@ -457,7 +457,10 @@ func saveDashboard(sess *db.Session, cmd *dashboards.SaveDashboardCommand, emitE
 
 		dash.UpdatedBy = userId
 
-		affectedRows, err = sess.MustCols("folder_id", "folder_uid").Nullable("folder_uid").ID(dash.ID).Update(dash)
+		// always clear deleted column on dashboard save
+		dash.Deleted = time.Time{}
```

Comment:
Wondering if we need to set deleted = null explicitly, instead of the zero value.
Null is what we look for on all non soft-deleted checks:
https://github.com/grafana/grafana/blob/a21a232a8e57eb1cd1f664806e60be4f38ddbca8/pkg/services/dashboards/database/database.go#L862

Reply:
yes, unfortunately we use a xorm's [Update()](https://xorm.io/docs/chapter-06/readme/) for the dashboard update and there is not at all obvious:
in order to set the column to NULL we have [to assign it to the zero value and mark the column as nullable](https://github.com/grafana/grafana/blob/fcfa4aa777dcd17836d81ea9dcf513bb95cc270f/pkg/util/xorm/session_update.go#L370)